.changeset/bright-eels-cross.md

```diff
@@ -0,0 +1,9 @@
+---
+'astro': minor
+---
+
+Exposes extra APIs for scripting and testing:
```

Comment:
Beautiful opening line, but then the rest doesn't sound HYPE enough for a minor, does it?? 😄 

1. Does "`build` API" refer to a new config reference item  `build.teardownCompiler`? If so, then we'll need to update the docs in this PR for config reference.

2. I'm looking for anywhere else we document things exported from `astro/config` and I guess it's just `defineCollection` that shows up in docs. So, what's the plan for people knowing about these and providing some kind of reference material?

Reply:
FYI `envField` is exported from `astro/config` too so I think for now it's just documented on the env guide

Reply:
I added the new functions as part of the experimental programmatic APIs since they are used in essentially the same context, doing parts of the CLI as a function.

Reply:
Sent a more detailed changeset